### ISSUE TITLE
After deploying, the chat app functions for me, but all other users are thrown a 500 error.

### ISSUE THREAD

**Issue by nclaunch** [microsoft/sample-app-aoai-chatGPT#590]
Consent is adding the users as expected, but they are redirected to a 500 error message. Any idea what the cause of this might be? Application was deployed through Azure AI Studio using the following article:

https://learn.microsoft.com/en-us/azure/ai-studio/tutorials/deploy-chat-web-app

**Comment by sarah-widder** [comment]
Hi @nclaunch can you share more about the 500 error? Perhaps network and console logs from the frontend, and logs from the backend?

**Comment by nclaunch** [comment]
Deleted the app, republished, and it self-corrected.

**Comment by nclaunch** [comment]
@sarah-widder when I'm redirected to https://xxx.azurewebsites.net/.auth/login/aad/callback after authentication, I get a 500 error. The logs available just show the 500 error without any real details.

**Comment by nclaunch** [comment]
After rebuilding the auth stack, the application now functions as expected. Thanks @sarah-widder for all your help!

**Comment by ill-yes** [comment]
I'm using openID and we're still facing this issue.

**Comment by sarah-widder** [comment]
@ill-yes Any luck from removing and re-adding the identity provider for your case? Can you share more details on exactly the configuration you have, and what behavior you're seeing?

**Comment by ill-yes** [comment]
I've readded it multiple times. Logs look like this:

```
2024-02-09T12:52:52.959112175Z Updated CA certificates
2024-02-09T12:52:57.207954345Z [41m[1m[37mcrit[39m[22m[49m: Microsoft.AspNetCore.Hosting.Diagnostics[6]
2024-02-09T12:52:57.207978545Z       Application startup exception
2024-02-09T12:52:57.207982745Z       System.TypeInitializationException: The type initializer for 'Microsoft.Azure.AppService.Middleware.ModuleUtils' threw an exception.
2024-02-09T12:52:57.207986845Z        ---> Newtonsoft.Json.JsonSerializationException: Error getting value from 'AuthorizationEndpointString' on 'Microsoft.Azure.AppService.Middleware.Modules.OpenIdConnectEndpointConfig'.
2024-02-09T12:52:57.207990945Z        ---> Microsoft.Azure.AppService.Middleware.HttpException: Exception of type 'Microsoft.Azure.AppService.Middleware.HttpException' was thrown.
2024-02-09T12:52:57.207994245Z          at Microsoft.Azure.AppService.Middleware.OpenIdConnectConfiguration.Download(String siteName, String url) in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/OpenIdConnectConfiguration.cs:line 77
2024-02-09T12:52:57.207997945Z          at Microsoft.Azure.AppService.Middleware.Modules.OpenIdConnectEndpointConfig.DownloadOpenIdConnectConfiguration(String wellKnownOpenIdConfigurationEndpoint) in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/Config/IdentityProviders/OpenIdConnectEndpointConfig.cs:line 240
2024-02-09T12:52:57.208001545Z          at Microsoft.Azure.AppService.Middleware.Modules.OpenIdConnectEndpointConfig.DownloadEndpointUrls() in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/Config/IdentityProviders/OpenIdConnectEndpointConfig.cs:line 229
2024-02-09T12:52:57.208005045Z          at lambda_method219(Closure , Object )
2024-02-09T12:52:57.208008045Z          at Newtonsoft.Json.Serialization.ExpressionValueProvider.GetValue(Object target)
2024-02-09T12:52:57.208011245Z          --- End of inner exception stack trace ---
2024-02-09T12:52:57.208014245Z          at Newtonsoft.Json.Serialization.ExpressionValueProvider.GetValue(Object target)
2024-02-09T12:52:57.208017445Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.CalculatePropertyValues(JsonWriter writer, Object value, JsonContainerContract contract, JsonProperty member, JsonProperty property, JsonContract& memberContract, Object& memberValue)
2024-02-09T12:52:57.208021745Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208025145Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208028445Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208031945Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208042845Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208046245Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208049645Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeDictionary(JsonWriter writer, IDictionary values, JsonDictionaryContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208053045Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208056645Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208060045Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208066245Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeObject(JsonWriter writer, Object value, JsonObjectContract contract, JsonProperty member, JsonContainerContract collectionContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208069645Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.SerializeValue(JsonWriter writer, Object value, JsonContract valueContract, JsonProperty member, JsonContainerContract containerContract, JsonProperty containerProperty)
2024-02-09T12:52:57.208072945Z          at Newtonsoft.Json.Serialization.JsonSerializerInternalWriter.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
2024-02-09T12:52:57.208076145Z          at Newtonsoft.Json.JsonSerializer.SerializeInternal(JsonWriter jsonWriter, Object value, Type objectType)
2024-02-09T12:52:57.208079246Z          at Newtonsoft.Json.JsonSerializer.Serialize(JsonWriter jsonWriter, Object value, Type objectType)
2024-02-09T12:52:57.208082246Z          at Newtonsoft.Json.JsonConvert.SerializeObjectInternal(Object value, Type type, JsonSerializer jsonSerializer)
2024-02-09T12:52:57.208085346Z          at Newtonsoft.Json.JsonConvert.SerializeObject(Object value, Type type, JsonSerializerSettings settings)
2024-02-09T12:52:57.208088446Z          at Newtonsoft.Json.JsonConvert.SerializeObject(Object value)
2024-02-09T12:52:57.208091446Z          at Microsoft.Azure.AppService.Middleware.EasyAuthModule.SetConfig(ConfigFactory configFactory, Boolean emitTelemetry) in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/EasyAuthModule.cs:line 122
2024-02-09T12:52:57.208094846Z          at Microsoft.Azure.AppService.Middleware.EasyAuthModule..ctor(ConfigFactory configFactory) in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/EasyAuthModule.cs:line 76
2024-02-09T12:52:57.208100446Z          at Microsoft.Azure.AppService.Middleware.ModuleUtils..cctor() in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/ModuleUtils.cs:line 52
2024-02-09T12:52:57.208103646Z          --- End of inner exception stack trace ---
2024-02-09T12:52:57.208106946Z          at Microsoft.Azure.AppService.Middleware.ModuleConfig.ForceLoadConfig() in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/ModuleConfig.cs:line 105
2024-02-09T12:52:57.208110146Z          at Microsoft.Azure.AppService.Middleware.ModuleConfig.EnsureConfigLoaded(HttpContextBase context, Boolean enableLegacyTracing, IConfiguration config) in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/ModuleConfig.cs:line 89
2024-02-09T12:52:57.208113546Z          at Microsoft.Azure.AppService.Middleware.NetCore.AppServiceMiddleware.InitializeConfig(ConfigurationHelper configurationHelper) in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.NetCore/AppServiceMiddleware.cs:line 48
2024-02-09T12:52:57.208116946Z          at Microsoft.Azure.AppService.Middleware.NetCore.AppServiceMiddleware..ctor(RequestDelegate next, ConfigurationHelper configHelper, Boolean isReverseProxy, Boolean isLocalHost) in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.NetCore/AppServiceMiddleware.cs:line 33
2024-02-09T12:52:57.208120346Z          at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
2024-02-09T12:52:57.208123746Z          at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
2024-02-09T12:52:57.208126946Z          at Microsoft.Extensions.Internal.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
2024-02-09T12:52:57.208130046Z          at Microsoft.Extensions.Internal.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
2024-02-09T12:52:57.208133246Z          at Microsoft.AspNetCore.Builder.UseMiddlewareExtensions.<>c__DisplayClass5_0.<UseMiddleware>b__0(RequestDelegate next)
2024-02-09T12:52:57.208136646Z          at Microsoft.AspNetCore.Builder.ApplicationBuilder.Build()
2024-02-09T12:52:57.208139746Z          at Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken)
2024-02-09T12:52:57.209623451Z Unhandled exception. System.TypeInitializationException: The type initializer for 'Microsoft.Azure.AppService.Middleware.ModuleUtils' threw an exception.
2024-02-09T12:52:57.209632651Z  ---> Newtonsoft.Json.JsonSerializationException: Error getting value from 'AuthorizationEndpointString' on 'Microsoft.Azure.AppService.Middleware.Modules.OpenIdConnectEndpointConfig'.
2024-02-09T12:52:57.209635951Z  ---> Microsoft.Azure.AppService.Middleware.HttpException: Exception of type 'Microsoft.Azure.AppService.Middleware.HttpException' was thrown.
2024-02-09T12:52:57.209638651Z    at Microsoft.Azure.AppService.Middleware.OpenIdConnectConfiguration.Download(String siteName, String url) in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/OpenIdConnectConfiguration.cs:line 77
2024-02-09T12:52:57.209646251Z    at Microsoft.Azure.AppService.Middleware.Modules.OpenIdConnectEndpointConfig.DownloadOpenIdConnectConfiguration(String wellKnownOpenIdConfigurationEndpoint) in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/Config/IdentityProviders/OpenIdConnectEndpointConfig.cs:line 240
2024-02-09T12:52:57.209649351Z    at Microsoft.Azure.AppService.Middleware.Modules.OpenIdConnectEndpointConfig.DownloadEndpointUrls() in /__w/1/s/src/EasyAuth/Microsoft.Azure.AppService.Middleware.Modules/Config/IdentityProviders/OpenIdConnectEndpointConfig.cs:line 229
2024-02-09T12:52:57.209652051Z    at lambda_method219(Closure , Object )
2024-02-09T12:52:57.209654551Z    at Newtonsoft.Json.Serialization.ExpressionValueProvider.GetValue(Object target)
2024-02-09T12:52:57.209657051Z    --- End of inner exception stack trace ---
2024-02-09T12:52:57.209659551Z    at Newtonsoft.Json.Serialization.ExpressionValueProvider.GetValue(Object target)
```

**Comment by seidnerj** [comment]
Am facing the same issue, suddenly I stared getting the following exception:

Newtonsoft.Json.JsonSerializationException: Error getting value from 'AuthorizationEndpointString' on 'Microsoft.Azure.AppService.Middleware.Modules.OpenIdConnectEndpointConfig'.

It clearly is something in the app service middle ware not able to parse the "well known configuration" JSON and fetch the "authorization endpoint" but for the life of me I don't know what the problem is. I've tried removing and re-adding the OpenID provider and honestly, nothing has changed, it just suddenly started throwing this exception. The worst thing is the app is down and the end user gets an http 500 code with no error - nada, all because the Azure middle ware fails. This is absurd.
